### DISCUSSION TITLE
Standalone executable

### DISCUSSION THREAD

**Discussion by CViniciusSDias** [skoro/php-tkui#41] [Q&A]
Unfortunatelly [phpmicro](https://github.com/easysoft/phpmicro) doesn´t seem to support FFI, so it's not possible to use it for generating standalone executables for desktop applications created with php-tkui.
Are there any alternatives?

**Comment by hsemix** [comment]
Static php cli

**Reply by CViniciusSDias** [reply]
That's the one I mentioned. phpmicro is built on top of that and add a few capabilities. Static php cli doesn't support FFI on linux.

**Reply by hsemix** [reply]
Oooh, you meant on linux, because I knew it supported it on at least two platforms.

**Reply by skoro** [reply]
Hmm... Looks like it's possible on [linux](https://static-php.dev/en/guide/extension-notes.html#ffi). Anyone tried ?

**Comment by socivdias** [comment]
Has anyone looked into TypePHP already for this?

Cc @skoro

**Reply by hsemix** [reply]
Well I think if there isn't much dynamic code, it should just work, just leave vendor alone

**Reply by socivdias** [reply]
What do you mean by "dynamic code" here? The lib heavily uses FFI, so...

**Reply by hsemix** [reply]
Things like eval(), $$, reflection and probably others
